### DISCUSSION TITLE
Command Cooldowns, Context vs App commands

### DISCUSSION THREAD

**Discussion by solumath** [Rapptz/discord.py#9704] [General] [answered]
Hello,
I've been wondering why the library doesn't have a united use of `command.cooldown` and `app_commands.check.cooldown`.
From what I see in the code the only problem should be the author attribute which is "user" in Interactions. To me it seems unnecessary to split these decorators. But I don't know anything about the implementation of discordpy library so can anybody explain the thought process and the reasoning?

**Comment by ika2kki** [comment, marked as answer]
hi. they're separate checks because they're implemented differently- `commands.cooldown` is handled in multiple places as part of the invoke flow whereas `app_commands.cooldown` is handled entirely within a check. something you might notice is `cooldown_after_parsing` is not supported in app commands because of this.
`ext.commands` is an extension as well and not a part of the core `discord` package
